### PR TITLE
Collections belong to users

### PR DESCRIPTION
**Pull request by Mentessi**
This feels like a small PR but a complete one (?) - collections do now belong to users and thats it (along with the tests and other minor changes). The next story will be something like "only collection owners can edit their own collections" and this feels like a separate thing so doing on its own branch.

### CODE REVIEW

**Comment by Mentessi** [comment]
a bit annoying I left this so long as progress was actually pretty smooth today and I think I've covered most bases (hopefully!).

File: app/controllers/collections_controller.rb [hunk 1]
```diff
@@ -43,8 +43,8 @@ def destroy
 
   private
 
-  def set_current_collection
-    @collection = Collection.find(params[:id])
+  def set_current_user_collection
```
Comment (Neilos) [inline app/controllers/collections_controller.rb]: Nice method naming 👍

File: app/controllers/topics_controller.rb [hunk 2]
```diff
@@ -21,7 +22,7 @@ def create
   end
 
   def show
-    @collections = Collection.all
+    @collections = Collection.where(user: current_user)
```
Comment (Neilos) [inline app/controllers/topics_controller.rb]: It works, but why not use your association methods:
```ruby
current_user.collections
```

File: db/migrate/20180611075005_add_collection_ref_to_users.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class AddCollectionRefToUsers < ActiveRecord::Migration[5.1]
```
Comment (Neilos) [inline db/migrate/20180611075005_add_collection_ref_to_users.rb]: Probably not worth fixing, but this migration name is misleading. It is really adding the user reference to the collections table.

File: db/migrate/20180611075005_add_collection_ref_to_users.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
...
+  def change
+    add_reference :collections, :user, foreign_key: true
```
Comment (Neilos) [inline db/migrate/20180611075005_add_collection_ref_to_users.rb]: Used the proper migration method. Nice one.

**Pull request merged**